### DISCUSSION TITLE
Internal Server Error on :4000 settings

### DISCUSSION THREAD

**Discussion by dabido3** [teslamate-org/teslamate#2110] [Q&A] [answered]
I'm not exactly when this error started happening, but I can't seem to access the :4000 settings page.

Running TM on a rpi with docker; data is logging up to date and viewable in grafana and my daily backups are occurring as intended.
I was up to date with v1.24.2 but tried to pull the version again just now, but still can't load the settings.

Any other details or help with troubleshooting would be appreciated :)

**Comment by zikronix** [comment]
same problem. Did the update, trying to hit the teslamate web interface produces the same error

**Reply by zikronix** [reply]
I updated docker and now things seem to be working

**Reply by dabido3** [reply]
You updated teslamate or updated docker itself?
(If the latter, How do you update docker?)

**Reply by zikronix** [reply]
> You updated teslamate or updated docker itself? (If the latter, How do you update docker?)

I had updated teslamate and thats when it broke. For updating docker, I ssh'd into the pi, and just updated everything on the pi using sudo apt update
sudo apt upgrade

**Reply by dabido3** [reply]
Pi full update fixed it!
Thanks

**Reply by Dellonb** [reply]
I had the same issue - Pi full update fixed it
thx

**Reply by natrlhy** [reply]
I did the same: `apt update;apt upgrade` and still get the Internal Server Error on the settings page...

```
# docker version
Client: Docker Engine - Community
 Version:           20.10.22
 API version:       1.41
 Go version:        go1.18.9
 Git commit:        3a2c30b
 Built:             Thu Dec 15 22:27:17 2022
 OS/Arch:           linux/arm64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.22
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.9
  Git commit:       42c8b31
  Built:            Thu Dec 15 22:25:39 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.15
  GitCommit:        5b842e528e99d4d4c1686467debf2bd4b88ecd86
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

**Comment by IMgoRt** [comment]
Just updated teslamate (docker-compose pull, up -d and now getting the same issue :4000 can't be reached
Been running this for 30 months with issue

**Comment by zikronix** [comment, marked as answer]
Update docker like I said above and see if that works

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by IMgoRt** [reply]
> Update docker like I said above and see if that works Get Outlook for iOS<https://aka.ms/o0ukef>
> […](#)
> ________________________________ From: IMgoRt ***@***.***> Sent: Saturday, October 30, 2021 11:24:31 AM To: adriankumpf/teslamate ***@***.***> Cc: zikronix ***@***.***>; Comment ***@***.***> Subject: Re: [adriankumpf/teslamate] Internal Server Error on :4000 settings (Discussion #2110) Just updated teslamate (docker-compose pull, up -d and now getting the same issue :4000 can't be reached Been running this for 30 months with issue — You are receiving this because you commented. Reply to this email directly, view it on GitHub<[#2110 (comment)](https://github.com/adriankumpf/teslamate/discussions/2110#discussioncomment-1563759)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AUVXJLCM3SKR5BOYQVIMBGLUJRA67ANCNFSM5HAAASDA>. Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

I should have mentioned that I'm on Synology and those functions aren't available

**Reply by IMgoRt** [reply]
> > Update docker like I said above and see if that works Get Outlook for iOShttps://aka.ms/o0ukef
> > […](#)
> > ________________________________ From: IMgoRt _**@**_._**> Sent: Saturday, October 30, 2021 11:24:31 AM To: adriankumpf/teslamate _**@**_.**_> Cc: zikronix _**@**_._**>; Comment _**@**_.**_> Subject: Re: [adriankumpf/teslamate] Internal Server Error on :4000 settings (Discussion #2110) Just updated teslamate (docker-compose pull, up -d and now getting the same issue :4000 can't be reached Been running this for 30 months with issue — You are receiving this because you commented. Reply to this email directly, view it on GitHub<[#2110 (comment)](https://github.com/adriankumpf/teslamate/discussions/2110#discussioncomment-1563759)>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AUVXJLCM3SKR5BOYQVIMBGLUJRA67ANCNFSM5HAAASDA. Triage notifications on the go with GitHub Mobile for iOShttps://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675 or Androidhttps://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub.
> 
> I should have mentioned that I'm on Synology and those functions aren't available

OK, restarted the docker subsystem without any improvement, restarted the server (that happens maybe once/year) and now all updated without issue

So nothing wrong with TM, something broken in my environment

**Comment by natrlhy** [comment]
In my case, I had extra entries in my `car_settings` table. After deleting the 2 extra that were loaner cars, I can now see my settings page again

**Reply by maracucho1997** [reply]
This fixed the issue for me! Looks like a rental car left an orphaned record in car_settings.

**Comment by bmrowe** [comment]
Seeing this same symptom. In my case, the error in the logs is:


```
request: GET /settings
** (exit) an exception was raised:
    ** (KeyError) key :id not found in: nil

If you are using the dot syntax, such as map.field, make sure the left-hand side of the dot is a map
        (teslamate 1.28.0-dev) lib/teslamate_web/live/settings_live/index.ex:183: anonymous fn/2 in TeslaMateWeb.SettingsLive.Index.prepare/1
        (elixir 1.15.6) lib/enum.ex:2510: Enum."-reduce/3-lists^foldl/2-0-"/3
        (teslamate 1.28.0-dev) lib/teslamate_web/live/settings_live/index.ex:15: TeslaMateWeb.SettingsLive.Index.mount/3
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/utils.ex:301: anonymous fn/6 in Phoenix.LiveView.Utils.maybe_call_live_view_mount!/5
        (telemetry 1.2.1) /opt/app/deps/telemetry/src/telemetry.erl:321: :telemetry.span/3
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/static.ex:270: Phoenix.LiveView.Static.call_mount_and_handle_params!/5
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/static.ex:110: Phoenix.LiveView.Static.render/3
        (phoenix_live_view 0.17.14) lib/phoenix_live_view/controller.ex:39: Phoenix.LiveView.Controller.live_render/3
```
        
        
From digging in the code a bit, I think this is because I sold a car that was at ID 1 (so I deleted it), and now the first car in the cars table starts with ID 2.   Is there some error handling missing here?

**Reply by bmrowe** [reply]
Found this thread - https://github.com/adriankumpf/teslamate/discussions/2719 

We definitely should build something to manage cars. Or make an FAQ on how to remove a car.

**Reply by cwanja** [reply]
I have thought about pushing a docs update to include how to remove a car. But I am not sure if Adrian is still support this after the incident. It seems abandoned.

**Comment by cleanev** [comment]
Team,

I had a loaner a while back that I removed once returned. Since then I updated once and migrated to new RPi once without any issues. Since I update to teslamate:teslamate:edge and now to the latest 1.27.4 I am unable to access settings page. I unfortunately created a new discussion [https://github.com/teslamate-org/teslamate/discussions/3412#discussion-5844438](url) that can be closed.

I did check all the suggestions in thread #2719 and its subsequent follow ups to no avail.
Any help will be appreciated.
